### PR TITLE
Implement eigen value calculator for symmetric matricies

### PR DESCRIPTION
**Pull request by unnonouno**
I implemented eigh and eigvals API.
It is copied from https://github.com/pfnet/chainer/pull/2429

### CODE REVIEW

**Comment by okuta** [comment]
LGTM except document.

**Comment by okuta** [comment]
Do you have plans to write documents?
If your answer is no, I will merge this PR.

**Comment by unnonouno** [comment]
I updated the document.

**Comment by okuta** [comment]
Thanks!
Do you want to add document for 'eigvalsh' and 'eigh'.

**Comment by unnonouno** [comment]
Oh, sorry. I did not write the document itself yet.

**Comment by unnonouno** [comment]
Currently it actually does not support Hermitian matrix because CuPy does not support complex numbers. Do i need to remove it from the description?

**Comment by okuta** [comment]
Yes, please. Please send another PR, that will merge after implementing complex numbers.

File: tests/cupy_tests/linalg_tests/test_eigenvalue.py [hunk 1]
```diff
@@ -1,9 +1,60 @@
...
 
+@unittest.skipUnless(
+    cuda.cusolver_enabled, 'Only cusolver in CUDA 8.0 is supported')
 @testing.gpu
 class TestEigenvalue(unittest.TestCase):
 
     _multiprocess_can_split_ = True
+
+    @testing.for_all_dtypes(no_float16=True)
+    @testing.numpy_cupy_allclose(rtol=1e-3, atol=1e-4)
```
Comment (unnonouno) [inline tests/cupy_tests/linalg_tests/test_eigenvalue.py]: Sorry, this is a bug. I fixed it.

**Comment by okuta** [comment]
LGTM!

**Pull request merged**